modules/db/src/main/java/org/opencastproject/db/Activator.java

```diff
@@ -198,6 +191,50 @@ public void start(BundleContext bundleContext) throws Exception {
 
   }
 
+  /**
+   * Attempts to establish a simple connection to the database. This method will retry if the connection attempt fails.
+   *
+   * @param jdbcDriver The JDBC driver class
+   * @param jdbcUrl The JDBC URL
+   * @param jdbcUser The JDBC user
+   * @param jdbcPass The JDBC password
+   */
+  private void waitUntilDatabaseIsAvailable(String jdbcDriver, String jdbcUrl, String jdbcUser, String jdbcPass) {
+    logger.info("Testing connectivity to database at {}", jdbcUrl);
+
+    // Load the JDBC driver for our basic connection test
+    try {
+      Class.forName(jdbcDriver);
+    } catch (ClassNotFoundException e) {
+      logger.error("Couldn't load JDBC Driver {} for initial connection test", jdbcDriver);
+      throw new RuntimeException(e);
```

Comment:
This is different from the connection settings we use above, Wouldn't it be better to use the same? In fact, couldn't you just put a loop around the `try-catch` block you removed above and modify the ctach block?

Reply:
Thank you for the review! Creating a loop around the original connection method was my first approach. However, I chose a different method for connecting to the database because the `pooledDataSource` logs exceptions upon failure. This alternative approach helps maintain a cleaner log. But we can use the `pooledDataSource` if the exceptions are acceptable.